### PR TITLE
HDDS-9269. [snapshot] Add unit-testcase for Snapshot Quota handling

### PR DESCRIPTION
**Pull request by jyotirmoy-gh**
## What changes were proposed in this pull request?

Added unit-testcase for Snapshot Quota handling - `TestOmSnapshot#testSnapshotQuotaHandling
`
## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-9269

## How was this patch tested?

Testcase file - `hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshot.java`
```
 mvn -l testlog.out -Dtest=TestOmSnapshot#testSnapshotQuotaHandling test
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.apache.hadoop.ozone.om.TestOmSnapshot
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 493.689 s - in org.apache.hadoop.ozone.om.TestOmSnapshot
[INFO]
[INFO] Results:
[INFO]
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0
[INFO]
```

### CODE REVIEW

**Comment by jyotirmoy-gh** [comment]
@sadanand48 @sumitagrawl @smengcl @nilotpalnandi - Please review

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshot.java [hunk 1]
```diff
@@ -1786,6 +1787,51 @@ public void testDeleteSnapshotMissingMandatoryParams() throws Exception {
...
+    assertEquals(buckSpaceQuotaBefore, buckSpaceQuotaAfter);
+
+    store.deleteSnapshot(volume, bucket, snap1);
+
+    long volNameQuotaFinal = volume1.getQuotaInNamespace();
+    long volSpaceQuotaFinal = volume1.getQuotaInBytes();
+    long buckNameQuotaFinal = bucket1.getQuotaInNamespace();
+    long buckSpaceQuotaFinal = bucket1.getQuotaInBytes();
+
+    assertEquals(volNameQuotaBefore, volNameQuotaFinal);
```
Comment (sadanand48) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshot.java:1829]: The Quota set on the bucket/volume doesn't change on taking /deleting a snapshot, we can do an assert for bucket.getUsedBytes() before and after these operations.

Comment (jyotirmoy-gh) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshot.java:1825]: Added checks for getUsedBytes and getUsedNamespace for volume and bucket.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshot.java [hunk 2]
```diff
@@ -1786,6 +1787,46 @@ public void testDeleteSnapshotMissingMandatoryParams() throws Exception {
...
+    OzoneVolume volume1 = store.getVolume(volume);
+    volume1.createBucket(bucket);
+    OzoneBucket bucket1 = volume1.getBucket(bucket);
+    bucket1.setQuota(OzoneQuota.parseQuota("102400000", "500"));
+    volume1.setQuota(OzoneQuota.parseQuota("204800000", "1000"));
+    // Create Key1 and take snapshot
+    String key1 = "key-1-";
+    createFileKeyWithPrefix(bucket1, key1);
+
+    long volNameQuotaBefore = volume1.getUsedNamespace();
```
Comment (sadanand48) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshot.java:1804]: volUsedNamespace

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshot.java [hunk 3]
```diff
@@ -1786,6 +1787,46 @@ public void testDeleteSnapshotMissingMandatoryParams() throws Exception {
...
+    long buckNameQuotaBefore = bucket1.getUsedNamespace();
```
Comment (sadanand48) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshot.java:1805]: buckUsedNamespace/buckNamespace

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshot.java [hunk 4]
```diff
@@ -1786,6 +1787,46 @@ public void testDeleteSnapshotMissingMandatoryParams() throws Exception {
...
+    long buckSpaceQuotaBefore = bucket1.getUsedBytes();
```
Comment (sadanand48) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshot.java:1806]: buckUsedBytes/buckSpace

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshot.java [hunk 5]
```diff
@@ -1786,6 +1787,46 @@ public void testDeleteSnapshotMissingMandatoryParams() throws Exception {
...
+
+    store.deleteSnapshot(volume, bucket, snap1);
+
+    long volNameQuotaFinal = volume1.getUsedNamespace();
+    long buckNameQuotaFinal = bucket1.getUsedNamespace();
+    long buckSpaceQuotaFinal = bucket1.getUsedBytes();
+
+    assertEquals(volNameQuotaBefore, volNameQuotaFinal);
+    assertEquals(buckNameQuotaBefore, buckNameQuotaFinal);
+    assertEquals(buckSpaceQuotaBefore, buckSpaceQuotaFinal);
```
Comment (sadanand48) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshot.java:1827]: maybe add a case where key is deleted after taking a snapshot

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshot.java (hunk 2 above)
Comment (jyotirmoy-gh) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshot.java]: Changed variable name

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshot.java (hunk 3 above)
Comment (jyotirmoy-gh) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshot.java]: Changed variable name

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshot.java (hunk 4 above)
Comment (jyotirmoy-gh) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshot.java]: Changed variable name

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshot.java (hunk 5 above)
Comment (jyotirmoy-gh) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshot.java]: Added delete scenario

**Review by sadanand48 — COMMENTED** [review]
Thanks @jyotirmoy-gh for the patch, LGTM.

**Review by sadanand48 — APPROVED** [review]

**Pull request merged**